milli/src/error.rs

```diff
@@ -211,6 +210,7 @@ impl fmt::Display for UserError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             Self::AttributeLimitReached => f.write_str("maximum number of attributes reached"),
+            Self::CriterionError(error) => f.write_str(&error.to_string()),
```

Comment:
Don't allocate and then write this allocation, prefer directly writing instead.

```suggestion
            Self::CriterionError(error) => write!(f, "{}", error),
```